ext/PathfinderTuringExt.jl

```diff
@@ -101,6 +103,58 @@ function varnames_to_ranges(metadata::DynamicPPL.Metadata)
     return Dict(zip(metadata.vns, ranges))
 end
 
+"""
+    transform_to_constrained(
+        p::AbstractArray, vi::DynamicPPL.VarInfo, model::DynamicPPL.Model
+    )
+
+Transform a vector of parameters `p` from unconstrained to constrained space.
+"""
+function transform_to_constrained(
+    p::AbstractArray, vi::DynamicPPL.VarInfo, model::DynamicPPL.Model
+)
+    p = copy(p)
+    @assert DynamicPPL.istrans(vi)
+    vi = DynamicPPL.unflatten(vi, p)
+    p .= DynamicPPL.invlink!!(vi, model)[:]
+    # Restore the linking, since we mutated vi.
+    DynamicPPL.link!!(vi, model)
+    return p
```

Comment:
I don't think we can use in-place inverse linking here, since the unconstrained parameter may not be of the same type/size as the constrained one (e.g. a `Dirichlet` or `LKJCholesky` parameter):
```suggestion
    p_trans = DynamicPPL.invlink(vi, model)[:]
    return p_trans
```

Reply:
That seems strictly better, I don't know why I did the acrobatics with `!!`.